projection/account_message/account_message.go

```diff
@@ -740,6 +740,147 @@ func (projection *AccountMessage) HandleEvents(height int64, events []event_enti
 					typedEvent.Params.Signer,
 				},
 			})
+		} else if typedEvent, ok := event.(*event_usecase.MsgGrant); ok {
+
+			var accounts []string
+
+			if typedEvent.Params.MaybeSendGrant != nil {
+				accounts = []string{
+					typedEvent.Params.MaybeSendGrant.Granter,
+					typedEvent.Params.MaybeSendGrant.Grantee,
+				}
+			} else if typedEvent.Params.MaybeStakeGrant != nil {
+				accounts = []string{
+					typedEvent.Params.MaybeStakeGrant.Granter,
+					typedEvent.Params.MaybeStakeGrant.Grantee,
+				}
+			} else if typedEvent.Params.MaybeGenericGrant != nil {
+				accounts = []string{
+					typedEvent.Params.MaybeGenericGrant.Granter,
+					typedEvent.Params.MaybeGenericGrant.Grantee,
+				}
+			}
+
+			accountMessage := view.AccountMessageRecord{
+				Row: view.AccountMessageRow{
+					BlockHeight:     height,
+					BlockHash:       "",
+					BlockTime:       utctime.UTCTime{},
+					TransactionHash: typedEvent.TxHash(),
+					Success:         typedEvent.TxSuccess(),
+					MessageIndex:    typedEvent.MsgIndex,
+					MessageType:     typedEvent.MsgType(),
+					Data:            typedEvent,
+				},
+				Accounts: accounts,
+			}
+
+			accountMessages = append(accountMessages, accountMessage)
+
+		} else if typedEvent, ok := event.(*event_usecase.MsgRevoke); ok {
+			accountMessages = append(accountMessages, view.AccountMessageRecord{
+				Row: view.AccountMessageRow{
+					BlockHeight:     height,
+					BlockHash:       "",
+					BlockTime:       utctime.UTCTime{},
+					TransactionHash: typedEvent.TxHash(),
+					Success:         typedEvent.TxSuccess(),
+					MessageIndex:    typedEvent.MsgIndex,
+					MessageType:     typedEvent.MsgType(),
+					Data:            typedEvent,
+				},
+				Accounts: []string{
+					typedEvent.Params.Granter,
+					typedEvent.Params.Grantee,
+				},
+			})
+		} else if typedEvent, ok := event.(*event_usecase.MsgExec); ok {
+			accountMessages = append(accountMessages, view.AccountMessageRecord{
+				Row: view.AccountMessageRow{
+					BlockHeight:     height,
+					BlockHash:       "",
+					BlockTime:       utctime.UTCTime{},
+					TransactionHash: typedEvent.TxHash(),
+					Success:         typedEvent.TxSuccess(),
+					MessageIndex:    typedEvent.MsgIndex,
+					MessageType:     typedEvent.MsgType(),
+					Data:            typedEvent,
+				},
+				Accounts: []string{
+					typedEvent.Params.Grantee,
```

Comment:
missing Granter here

Reply:
There is no granter in this message 😅 

https://docs.cosmos.network/master/modules/authz/03_messages.html#msgexec

Reply:
so there might be a possible issue to lose Granter info in the inner message. Or do the inner messages contain the Granter address?

Reply:
@ysong42 did we confirm the `Granter` field yet?

Reply:
@davcrypto Sorry missed your message 🙏 

Yes, it is possible to lose Granter info in the inner message.

However, the inner messages should always contain the `Granter` address. Otherwise, there is no need to wrap a message inside the MsgExec.

The Granter address could be store under different fields in the inner message.
If the inner message is `MsgSend`, the `Granter` address should be the value `fromAddress`.
If the inner message is `MsgDeposit`, the `Granter` address should be the value `depositor `.
If the inner message is `MsgVote`, the `Granter` address should be the value `voter`.

Reply:
As in another PR #526 , I will parse out the inner messages and they will handled by HandleEvents separately. Therefore, I think we do not need to handle the inner message explicitly here.

Reply:
do you think we need a relationship identifier for those inner msgs and the exec msg?

Reply:
I think we won't need it. As the inner msgs and the exec msg still under the same transactions. One could check that specific transaction to find the MsgExec and inner messages.